src/renderer/hooks/useMapServerQuery.js

```diff
@@ -0,0 +1,15 @@
+// @ts-check
+import ky from 'ky/umd'
+import { useQuery } from '@tanstack/react-query'
+
+// local host and global port number
+// @ts-ignore
+const MAP_SERVER_URL = 'http://127.0.0.1:' + window.mapServerPort
+
+export function useMapServerQuery (resourcePath, enabled) {
```

Comment:
wondering if the idiomatic approach to using react-query is to create a query for each kind of resource. for example instead of the `resourcePath` determining what resource we're working with, it could be:

```js
function useGetStylesQuery(enabled) {
  return useQuery(['styles'], ...)
}

function useGetStyleQuery(styleId, enabled) {
  return useQuery(['styles', styleId], ...)
}

// etc...
```

Again, not super familiar with react query best practices but a quick look at the docs seems to lean towards this approach

Reply:
Yeah, Gregor and I talked a little about this. We were thinking that we did not want to create too many abstractions, given the simplicity of the API. So we decided to just use the one hook, and have the resource path as the keys. I think when we do more complicated things in the future this would be the way to go.

Reply:
got it, figured it was discussed but asking for my own education whenever i eventually touch this stuff 😄